File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;

        queue<TreeNode*> data;
        data.push(root);

        while(!data.empty()) {
            vector<int> level;

            for (int i = data.size(); i > 0; i--) {
                TreeNode* node = data.front();
                data.pop();

                if (node) {
                    level.push_back(node->val);
                    data.push(node->left);
                    data.push(node->right);
                }
            }

            if(!level.empty()) {
                result.push_back(level);
            }
        }

        return result;
    }
};

File: Data Structures & Algorithms/remove-all-adjacent-duplicates-in-string-ii/submission-0.py
class Solution:
    def removeDuplicates(self, s: str, k: int) -> str:
        stack = []
        for val in s:
            if stack and stack[-1][0] == val:
                stack[-1][1] += 1
            else:
                stack.append([val, 1])
            if stack[-1][1] == k:
                stack.pop()
        result = ''
        for item in stack:
            result += item[0] * item[1]
        return result